packages/@expo/cli/e2e/__tests__/export/static-middleware.test.ts

```diff
@@ -0,0 +1,36 @@
+/* eslint-env jest */
+import path from 'path';
+
+import { runExportSideEffects } from './export-side-effects';
+import { executeExpoAsync } from '../../utils/expo';
+import { getRouterE2ERoot } from '../utils';
+
+runExportSideEffects();
+
+describe('static export with middleware', () => {
+  const projectRoot = getRouterE2ERoot();
+  const outputName = 'dist-static-middleware-async';
+  const outputDir = path.join(projectRoot, outputName);
+
+  beforeAll(async () => {
+
+  });
+
+  describe('static', () => {
+    it('skips middleware when exporting a project with web.output === static', async () => {
+      const results = await executeExpoAsync(
+        projectRoot,
+        ['export', '-p', 'web', '--output-dir', outputName],
+        {
+          env: {
+            NODE_ENV: 'production',
+            EXPO_USE_STATIC: 'static',
+            E2E_ROUTER_SRC: 'server-middleware-async',
+          },
+        }
+      );
+
+      expect(results.stderr).toContain('Skipping export for middleware because `web.output` is not "server"')
```

Comment:
Would it make sense to also check that the file was not created in the exported files?